graphql/src/datasources/es.ts

```diff
@@ -250,17 +258,36 @@ class ElasticSearchAPI extends RESTDataSource {
       query.query.bool.filter = filters;
     }
 
-    if (typeof orderByDistance !== 'undefined') {
-      query.sort = {
-        _geo_distance: {
-          location: {
-            lat: orderByDistance.latitude,
-            lon: orderByDistance.longitude,
+    const hasDistanceOrdering = typeof orderByDistance !== 'undefined';
+    const hasNameOrdering = typeof orderByName !== 'undefined';
+
+    if (hasDistanceOrdering && hasNameOrdering) {
+      throw new ValidationError(
+        'Cannot use both orderByDistance and orderByName.'
+      );
+    }
+    if (['event', 'location'].includes(es_index)) {
+      if (hasDistanceOrdering) {
+        query.sort = {
+          _geo_distance: {
+            location: {
+              lat: orderByDistance.latitude,
+              lon: orderByDistance.longitude,
+            },
+            order: orderByDistance.order === 'DESCENDING' ? 'desc' : 'asc',
+            ignore_unmapped: true,
           },
-          order: orderByDistance.order === 'DESCENDING' ? 'desc' : 'asc',
-          ignore_unmapped: true,
-        },
-      };
+        };
+      } else if (hasNameOrdering) {
+        query.sort = {
+          [`${es_index === 'location' ? 'venue' : 'event'}.name.${
+            languages[0] ?? 'fi'
+          }.keyword`]: {
```

Comment:
I think this is little bit hard to read when the whole thing is inlined. Would somehting like this be easier?

```
const type = es_index === 'location' ? 'venue' : 'event';
const language = languages[0] ?? 'fi';
query.sort = {
  [`${type}.name.${language}.keyword`]: {
    order: orderByName.order === 'DESCENDING' ? 'desc' : 'asc',
    missing: '_last',
  },
};
```